worldedit-core/src/main/java/com/sk89q/worldedit/LocalSession.java

```diff
@@ -405,6 +406,15 @@ public void setTimezone(ZoneId timezone) {
      */
     public void clearHistory() {
         //FAWE start
+        if (Fawe.isMainThread() && !historyWriteLock.tryLock()) {
```

Comment:
I think this `tryLock()` requires a matching `unlock()`, otherwise the hold count won't go down to 0 and actually unlock it.

Reply:
Using a reentrant lock shouldn't require that? I don't think it has a concept of lock counts

Reply:
Ah okay, fair then